bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,839 @@
+                if (result != null && !result.isFailed()) {
+                    final String response = getContentAsString();
+                    if (response != null) {
+                        scheduler.execute(() -> {
+                            logger.trace("Processing response");
+                            if (daily) {
+                                lastDailyResponse = response;
+                                pollForDataDailyData(lastDailyResponse);
+                            } else {
+                                lastHourlyResponse = response;
+                                pollForDataHourlyData(lastHourlyResponse);
+                            }
+                        });
+                        if (authFailedPreviously && getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                            updateStatus(ThingStatus.ONLINE);
+                        }
+                    }
+                } else {
+                    logger.debug("Failed to get latest MET office forecast");
+                    // Clear the web servers thread
+                    if (result != null) {
+                        final Response resp = result.getResponse();
+                        if (resp != null) {
+                            scheduler.execute(() -> {
+                                handleNon200Response(resp);
+                            });
+                        }
+                    }
+                }
+            }
+        };
+
+        sendAsyncSiteApiRequest(httpClient, daily, uplinkBridge.getApiKey(), location, listener);
+    }
+
+    protected static void sendAsyncSiteApiRequest(final HttpClient httpClient, final boolean daily, final String apiKey,
+            final PointType location, final Response.CompleteListener listener) {
+        final String url = ((daily) ? GET_FORECAST_URL_DAILY : GET_FORECAST_URL_HOURLY)
+                .replace("<LATITUDE>", location.getLongitude().toString())
+                .replace("<LONGITUDE>", location.getLongitude().toString());
+
+        final Request request = httpClient.newRequest(url).method(HttpMethod.GET)
+                .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_TYPE.toString())
+                .header(GET_FORECAST_API_KEY_HEADER, apiKey).timeout(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        request.send(listener);
+    }
+
+    private static String calculatePrefix(final String prefix, final int plusOffset) {
+        final StringBuilder strBldr = new StringBuilder(26);
+        strBldr.append(prefix);
+        if (plusOffset > 0) {
+            strBldr.append(GROUP_POSTFIX_BOTH_FORECASTS);
+            if (plusOffset < 10) {
+                strBldr.append("0");
+            }
+            strBldr.append(plusOffset);
+        }
+        strBldr.append(GROUP_PREFIX_TO_ITEM);
+        return strBldr.toString();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
```

Comment:
Guess you want to reset these to the default state as the biniding (re)initilizes
```suggestion
        requiresHourlyData = false
        requiresDailyData = false
```

Reply:
Yeah the whole re-initialization is being re-tested in depth at the moment, hopefully this is no longer applicable.